### PR TITLE
subiquity_client: add OsProber

### PR DESCRIPTION
**Pull request by jpnurmi**
Ref: #428

### CODE REVIEW

File: packages/subiquity_client/lib/src/types.dart [hunk 1]
```diff
@@ -196,14 +196,29 @@ enum ProbeStatus { PROBING, FAILED, DONE }
 
 enum Bootloader { NONE, BIOS, UEFI, PREP }
 
+@freezed
+class OsProber with _$OsProber {
+  @JsonSerializable(fieldRename: FieldRename.snake)
+  const factory OsProber({
```
Comment (jpnurmi) [inline packages/subiquity_client/lib/src/types.dart:202]: P.S. Missing coverage for this const constructor is a bug in Dart: https://github.com/flutter/flutter/issues/95008 (already fixed in the latest Flutter master)

File: packages/subiquity_client/test/types_test.dart [hunk 2]
```diff
@@ -115,6 +115,25 @@ void main() {
...
+      'type': 'efi',
+      'subpath': '/efi/Microsoft/Boot/bootmgfw.efi',
+      'version': '11',
+    };
+    expect(os.toJson(), equals(json));
+    expect(OsProber.fromJson(json), os);
+  });
+
   test('partition', () {
     const keyboard = Partition(
```
Comment (oSoMoN) [inline packages/subiquity_client/test/types_test.dart:138]: Not strictly related to this change, but the name of the test object is misleading, it should be `partition`, not `keyboard`. Probably a copy-paste leftover from another test…

**Review by oSoMoN — APPROVED** [review]
LGTM, see just one minor comment for something unrelated to this change.

File: packages/subiquity_client/test/types_test.dart (hunk 2 above)
Comment (jpnurmi) [reply packages/subiquity_client/test/types_test.dart]: i hope it's fine as a separate commit

Comment (oSoMoN) [reply packages/subiquity_client/test/types_test.dart]: Yes, that's actually better, thanks!

**Pull request merged**